sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorHandlerProvider.java

```diff
@@ -34,9 +39,24 @@ public class ReactorHandlerProvider {
      * @param provider The provider that creates and manages {@link Reactor} instances.
      *
      * @throws NullPointerException If {@code provider} is {@code null}.
+     * @deprecated use {@link ReactorHandlerProvider#ReactorHandlerProvider(ReactorProvider, Meter)} instead.
      */
+    @Deprecated
     public ReactorHandlerProvider(ReactorProvider provider) {
+        this(provider, null);
+    }
```

Comment:
Because this is non-public API, and if you want people to always use the other API, you could consider just removing this constructor.

Reply:
EventHubs and ServiceBus will call into this constructor, Since we can never guarantee version alignment between azure-core-amqp and client libs,  I'd really like to keep this constructor even after SB/EH start using it for a while to minimize version conflicts